src/indexer/index.ts

```diff
@@ -230,3 +231,35 @@ export function decodeStateMap(state: Array<any>): Map<Key, StateValue> {
 	}
 	return stateMap;
 }
+
+export function decodeAppParamsState(state: any): Map<Key, StateValue> {
+	const stateMap = new Map<Key, StateValue>();
+	if (state) {
+		for (const key in state) {
+			if (state[key].at === SchemaType.BYTES) {
+				stateMap.set(
+					Buffer.from(key, "base64").toString("ascii"),
+					Buffer.from(state[key].bs, "base64").toString("ascii")
+				);
+			} else {
+				stateMap.set(
+					Buffer.from(key, "base64").toString("ascii"),
+					state[key].ui
+				);
+			}
+		}
+	}
+	return stateMap;
+}
+
+export async function getAssetInformation(
+	assetId: number
+): Promise<UnknownObject> {
+	try {
+		const assetInfo = await indexerClient.lookupAssetByID(assetId).do();
```

Comment:
Shouldn't we check if the value is present in it before? Array gives runtime error if the index value is not present.

Reply:
It will exist since it is there in the dao table that means it is deployed, also we are not checking for account specific asset data rather we are fetching for asset info itself

Reply:
Ok